rollbar/test/test_rollbar.py

```diff
@@ -117,6 +117,379 @@ def test_wsgi_request_data(self):
+        from rollbar.contrib.fastapi import ReporterMiddleware
+
+        app = FastAPI()
+        app.add_middleware(ReporterMiddleware)
+
+        # Inject annotations and decorate endpoint dynamically
+        # to avoid SyntaxError for older Python
+        #
+        # @app.get('/{param}')
+        # def root(param, fastapi_request: Request):
+        #     current_request = rollbar.get_request()
+        #
+        #     self.assertEqual(current_request, fastapi_request)
+
+        def root(param, fastapi_request):
+            current_request = rollbar.get_request()
+
+            self.assertEqual(current_request, fastapi_request)
+
+        root = fastapi_add_route_with_request_param(
+            app, root, '/{param}', 'fastapi_request'
+        )
+
+        client = TestClient(app)
+        response = client.get('/test?param1=value1&param2=value2')
+
+        self.assertEqual(response.status_code, 200)
+
+    @unittest.skipUnless(sys.version_info >= (3, 7), 'Python3.7+ required')
+    def test_get_request_fastapi_logger(self):
+        try:
+            from fastapi import FastAPI, Request
+            from fastapi.testclient import TestClient
+        except ImportError:
+            self.skipTest('Requires FastaAPI package')
+        from rollbar.contrib.fastapi import ReporterMiddleware
+
+        app = FastAPI()
+        app.add_middleware(ReporterMiddleware)
+
+        # Inject annotations and decorate endpoint dynamically
+        # to avoid SyntaxError for older Python
+        #
+        # @app.get('/{param}')
+        # def root(fastapi_request: Request):
+        #     current_request = rollbar.get_request()
+        #
+        #     self.assertEqual(current_request, fastapi_request)
+
+        def root(param, fastapi_request):
+            current_request = rollbar.get_request()
+
+            self.assertEqual(current_request, fastapi_request)
+
+        root = fastapi_add_route_with_request_param(
+            app, root, '/{param}', 'fastapi_request'
+        )
+
+        client = TestClient(app)
+        response = client.get('/test?param1=value1&param2=value2')
+
+        self.assertEqual(response.status_code, 200)
+
+    @unittest.skipUnless(sys.version_info >= (3, 7), 'Python3.7+ required')
+    def test_get_request_fastapi_router(self):
+        try:
+            import fastapi
+            from fastapi import FastAPI, Request
+            from fastapi.testclient import TestClient
+        except ImportError:
+            self.skipTest('Requires FastAPI package')
+        from rollbar.contrib.fastapi import add_to as rollbar_add_to
+
+        if fastapi.__version__ < '0.41.0':
+            self.skipTest('Requires FastAPI 0.41.0+')
+
+        app = FastAPI()
+        rollbar_add_to(app)
+
+        # Inject annotations and decorate endpoint dynamically
+        # to avoid SyntaxError for older Python
+        #
+        # @app.get('/{param}')
+        # def root(fastapi_request: Request):
```

Comment:
The purpose of the commented out code is not clear to me, maybe expand the comment, or remove the code.

Reply:
Take a look at the b4c6edb3 commit for a better picture.

The purpose of the commented out code is to show the actual test/use case that had to be worked around due to Python version incompatibilities. This code should be uncommented after the upcoming general test refactoring.

Please suggest how to expand the comment if it's not descriptive enough.

Reply:
So does the commented out code represent what you'd use if you didn't need to support Python < 3.5? If that's the case what I would expect something like 

```
        # Inject annotations and decorate endpoint dynamically
        # to avoid SyntaxError for older Python
        # This is the code we'd use if we could limit support to Python >=3.5
```

Reply:
Not completely. This test will never run on Python < 3.6 as FastAPI requires 3.6+ and we don't intention to back port it. However the SyntaxError is raised only for Python2 (annotations are supported in Py3+).

The actual problem is with the test suite and test runner. Note a decorator for the test `@unittest.skipUnless(sys.version_info >= (3, 7), 'Python3.7+ required')`.

However, we would need to move the test out of the file (to not physically import it under Python2) or reorganize tests and use better test runner. It's on our roadmap but not as a part of this integration.

How about this?

```
        # Inject annotations and decorate endpoint dynamically
        # to avoid SyntaxError for older Python
        # This is the code we'd use if we had not loaded the test file on Python 2.
```

Reply:
:+1: